source/_docs/composer-drupal-8.md

```diff
@@ -0,0 +1,130 @@
+
+```
+composer config repositories.drupal composer https://packagist.drupal-composer.org
+```
+
+Once Composer knows where to find modules, they can be added easily with `composer require`. For instance, [address module](https://www.drupal.org/project/address) could be installed via Composer with
+
+```
+composer require drupal/address
+
+```
+
+Address module depends on a few external packages (because problems of mailing addresses are not Drupal-specific). Those external packages are downloaded to the `vendor` directory. Address module itself will be put in Drupal's `module` directory. `composer require` can also be used for themes and profiles which will also be placed in the appropriate Drupal directories.
+
+
+## Whether to commit the vendor directory
+
+[Much of the documentation on Composer says that the `vendor` directory should not be committed to version control.](https://getcomposer.org/doc/faqs/should-i-commit-the-dependencies-in-my-vendor-directory.md) This recommendation presumes some abstraction layer between the git repository developers commit to and the way their code is deployed to a live site (and test sites for that matter). That abstraction might be a build system that takes the development team's repository, and creates a "build artifact" from the repository (by running `composer install` and possibly other steps like compiling Sass) which is then deployed. That build artifact might be tracked in a separate git repository. [In fact, this mental model has been the way many teams have used Pantheon for years](https://pantheon.io/blog/example-repository-build-drupal-composer-travis). They use GitHub, Bitbucket, or some other external repository as the repository where developers commit their changes. A continuous integration service then builds the developers' changes and sends them to Pantheon. If this is the model you are using, you do not have to commit your `vendor` directory.
+
+If Pantheon is your site's only repository, then you do have to commit the `vendor` directory. Pantheon uses the same git repository to control how code is deployed to all Pantheon environments. We consider it a security feature to make it easy for developers to see how exactly the same code is deployed to all environments. Many teams using Pantheon consider a two-repository model to be unnecessarily complex.
+
+### Avoiding committing dependencies as submodules
+
+**@todo, recommend one of the for approaches listed in https://getcomposer.org/doc/faqs/should-i-commit-the-dependencies-in-my-vendor-directory.md The gitignore option seems like the one that would work cleanest on Pantheon and could be changed in drops-8. However I have not been able to get it to work. In using `vendor/**/.git` as a .gitignore rule, I still get packages added as submodules.**
```

Comment:
So, I tried adding `/vendor/**/.git` to .gitignore, and that actually did work for me. However, it does not work if you run `git add .` before adding the item to .gitignore.  `git reset .` was enough to get me out of this trouble, though. If you have committed the submodule prior to adding the entry to .gitignore, then you need to use `git rm` per http://git.661346.n2.nabble.com/quot-submodule-quot-mistake-and-a-problem-td7568253.html